### PR TITLE
Manager upgrade

### PR DESCRIPTION
**Pull request by yarongilor**
1) Added upgrade support for scylla manager
```
* 40ec9f5 - Added support for Manager upgrade in 'upgrade_mgmt()'.
* eff5a28 - Added Manager upgrade test: test_manager_upgrade.
```
2) Added fix to test_mgmt_repair_nemesis that should be backported to both manager-1.2 and manager-1.3 branches.
```
* 05fcebb - (HEAD -> manager_upgrade, origin/manager_upgrade) Fixed test_mgmt_repair_nemesis to run MgmtRepair nemesis as test object instead of background thread.
```
3) test_manager_upgrade sample output is:
```
2018-12-05 17:57:44,232 mgmt             L0483 DEBUG| Issuing: 'sctool version'
2018-12-05 17:57:46,197 remote           L0194 DEBUG| [54.89.206.90] [stdout] Client version: 1.2.0-0.20181011.10a1d52
2018-12-05 17:57:46,223 remote           L0194 DEBUG| [54.89.206.90] [stdout] Server version: 1.2.0-0.20181011.10a1d52
2018-12-05 17:57:46,226 mgmt_cli_test    L0145 INFO | Running Manager upgrade from: [['Client version: 1.2.0-0.20181011.10a1d52'], ['Server version: 1.2.0-0.20181011.10a1d52']] to: http://downloads.scylladb.com/manager/rpm/unstable/centos/branch-1.3/5/scylla-manager.repo
2018-12-05 17:57:48,254 cluster          L1113 DEBUG| Node yaron_manager_multidc-monitor-node-7d927574-1 [54.89.206.90 | 172.30.0.18] (seed: None): Upgrade scylla-manager via repo: http://downloads.scylladb.com/manager/rpm/unstable/centos/branch-1.3/5/scylla-manager.repo
2018-12-05 17:57:51,363 remote           L0194 DEBUG| [54.89.206.90] [stdout] Upgrade  1 Package (+2 Dependent packages)
2018-12-05 17:57:58,191 mgmt             L0483 DEBUG| Issuing: 'sctool version'
2018-12-05 17:58:00,047 remote           L0194 DEBUG| [54.89.206.90] [stdout] Client version: 1.3.0-0.20181205.89dd4a6
2018-12-05 17:58:00,067 remote           L0194 DEBUG| [54.89.206.90] [stdout] Server version: 1.3.0-0.20181205.89dd4a6
2018-12-05 17:58:00,070 mgmt_cli_test    L0147 INFO | The Manager version after upgrade is: [['Client version: 1.3.0-0.20181205.89dd4a6'], ['Server version: 1.3.0-0.20181205.89dd4a6']]
yarongilor@yaron-pc:~/avocado/job-results/latest$ 
```

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1134,21 +1134,33 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
...
+                            apt-get install dirmngr
+                            apt-key adv --fetch-keys https://download.opensuse.org/repositories/home:/scylladb:/scylla-3rdparty-jessie/Debian_8.0/Release.key
                                     """)
-            self.remoter.run('sudo bash -cxe "%s"' % install_open_jdk)
+            self.remoter.run('sudo bash -cxe "%s"' % install_debian_9_prereqs)
+
+        install_open_jdk = dedent("""
+                                    apt-get install -y openjdk-8-jre-headless
+                                    update-java-alternatives --jre-headless -s java-1.8.0-openjdk-amd64
+                                            """)
```
Comment (bentsi) [inline sdcm/cluster.py]: indentation

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1134,21 +1134,33 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
...
             self.remoter.run('sudo bash -cxe "%s"' % install_transport_https)
-            install_open_jdk = dedent("""
-                             sudo apt-get install -t jessie-backports ca-certificates-java -y
-                            sudo apt-get install -y openjdk-8-jre-headless
-                            sudo update-java-alternatives --jre-headless -s java-1.8.0-openjdk-amd64
+
+
+        if self.is_debian9():
+            install_debian_9_prereqs = dedent("""
+                            if [ ! -f /etc/apt/sources.list.d/backports.list ]; then sudo echo 'deb http://http.debian.net/debian jessie-backports main' | sudo tee /etc/apt/sources.list.d/backports.list > /dev/null; fi
```
Comment (bentsi) [inline sdcm/cluster.py]: no need in `sudo`, the whole script is run under `sudo`

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1134,21 +1134,33 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
...
+                apt-get install -t jessie-backports ca-certificates-java -y
                         """)
             self.remoter.run('sudo bash -cxe "%s"' % install_transport_https)
-            install_open_jdk = dedent("""
-                             sudo apt-get install -t jessie-backports ca-certificates-java -y
-                            sudo apt-get install -y openjdk-8-jre-headless
-                            sudo update-java-alternatives --jre-headless -s java-1.8.0-openjdk-amd64
+
+
+        if self.is_debian9():
```
Comment (bentsi) [inline sdcm/cluster.py]:
```
if 
elif ...
elif ...
else:
     raise Exception("Unsupported OS")
```

File: sdcm/cluster.py [hunk 4]
```diff
@@ -2493,10 +2505,14 @@ def install_scylla_monitoring_prereqs(self, node):
             """)
         else:
             prereqs_script = dedent("""
+                apt-get update
```
Comment (bentsi) [inline sdcm/cluster.py]: As discussed `else` should raise exception and this script is for Debian

File: sdcm/cluster.py (hunk 1 above)
Comment (yarongilor) [reply sdcm/cluster.py]: done

File: sdcm/cluster.py (hunk 2 above)
Comment (yarongilor) [reply sdcm/cluster.py]: done

File: sdcm/cluster.py (hunk 3 above)
Comment (yarongilor) [reply sdcm/cluster.py]: in this case added a preceding condition of:
if not (self.is_rhel_like() or self.is_debian() or self.is_ubuntu())

File: sdcm/cluster.py (hunk 4 above)
Comment (yarongilor) [reply sdcm/cluster.py]: done

**Pull request merged**